src/ngraph/node.cpp

```diff
@@ -485,18 +486,28 @@ std::tuple<element::Type, PartialShape> Node::validate_and_infer_elementwise_arg
                 element::Type::merge(element_type, element_type, get_input_element_type(i)),
                 "Argument element types are inconsistent.");
 
-            NODE_VALIDATION_CHECK(this,
-                                  PartialShape::merge_into(pshape, get_input_partial_shape(i)),
-                                  "Argument shapes are inconsistent.");
+            if (autob == op::AutoBcastType::NUMPY)
+            {
+                NODE_VALIDATION_CHECK(
+                    this,
+                    PartialShape::bcast_merge_into(pshape, get_input_partial_shape(i), autob),
+                    "Argument shapes are inconsistent.");
+            }
+            else
```

Comment:
Might want to change this to `else if(autob==NONE)` and have an `else` that catches not-yet-implemented broadcast schemes.

Reply:
I was thinking along the lines of a broadcast resolver pure class that frameworks implement. Core could provide a NumPy implementation as an example, although I think none of our frameworks actually do NumPy.

Reply:
Not sure I like the idea of an opaque resolver object. With that we wouldn't be able to deserialize graphs with implicit broadcast and run them in `nbench`, for example.